src/Func.h

```diff
@@ -2281,6 +2281,12 @@ class Func {
      */
     Func &async();
 
+    /** Adds an extra dimension to the function storage to enable
+     * double buffering. This is especially useful in combination
+     * with the \ref Func::async.
+     */
+    Func &ring_buffer(Expr extent);
```

Comment:
This needs an explanation on where that dimension is gonna get inserted. We are introducing ring_buffering, but how does it know over which loop?